src/realm/sync/instruction_applier.cpp

```diff
@@ -114,6 +142,7 @@ void InstructionApplier::operator()(const Instruction::CreateObject& instr)
 {
     auto table = get_table(instr);
     ColKey pk_col = table->get_primary_key_column();
+    m_last_object_key = instr.object;
```

Comment:
I notice that this is a change that will result in one less object lookup for instruction sets that use create -> update. But that should be well tested already.